luigi/worker.py

```diff
@@ -108,10 +108,10 @@ def __init__(self, task, worker_id, result_queue, tracking_url_callback,
         self.result_queue = result_queue
         self.tracking_url_callback = tracking_url_callback
         self.status_message_callback = status_message_callback
-        self.random_seed = random_seed
         if task.worker_timeout is not None:
             worker_timeout = task.worker_timeout
         self.timeout_time = time.time() + worker_timeout if worker_timeout else None
+        self.random_seed = random_seed or self.timeout_time is not None
```

Comment:
* What is `random_seed`?
 * Is there any code-path where it's not set?
 * Is there any code-path but [this one](https://github.com/Houzz/luigi/blob/afe01d32eb8a7fac1249554a8a658393a66a530e/luigi/worker.py#L149-L151) where it's used?

To me it seems like we can remove it entirely. For a moment I thought it could be a `TaskProcess` thingie, but I only found luigi-related results when [googling](https://www.google.com/search?client=ubuntu&channel=fs&q=multiprocess.Process+random_seed&ie=utf-8&oe=utf-8) it.